scripts/isolate_tests.py

```diff
@@ -59,48 +91,44 @@ def extract_docs_cases(path):
             if line == '' or line.startswith(" "):
                 tests[-1] += line + "\n"
                 immediatelyAfterMarker = False
-            else:
-                insideBlock = False
-        elif any(map(line.lower().startswith, beginMarkers)):
+                continue
+
+            insideBlock = False
+        if any(map(line.lower().startswith, beginMarkers)):
             insideBlock = True
             immediatelyAfterMarker = True
             tests += ['']
 
-    codeStart = "(// SPDX-License-Identifier:|pragma solidity|contract.*{|library.*{|interface.*{)"
-
-    for test in tests:
-        if re.search(r'^\s{0,3}' + codeStart, test, re.MULTILINE):
-            print("Indentation error in " + path + ":")
-            print(test)
-            exit(1)
-
-    # Filter out tests that are not supposed to be compilable.
-    return [
-        test.lstrip("\n")
-        for test in tests
-        if re.search(r'^\s{4}' + codeStart, test, re.MULTILINE) is not None
-    ]
+    return tests
 
-def write_cases(f, tests):
+def write_cases(f, solidityTests, yulTests):
     cleaned_filename = f.replace(".","_").replace("-","_").replace(" ","_").lower()
-    for test in tests:
+    for language, test in [("sol", t) for t in solidityTests] + [("yul", t) for t in yulTests]:
         # When code examples are extracted they are indented by 8 spaces, which violates the style guide,
         # so before checking remove 4 spaces from each line.
-        remainder = re.sub(r'^ {4}', '', test, 0, re.MULTILINE)
-        sol_filename = 'test_%s_%s.sol' % (hashlib.sha256(test.encode("utf-8")).hexdigest(), cleaned_filename)
+        remainder = dedent(test)
+        sol_filename = 'test_%s_%s.%s' % (hashlib.sha256(test.encode("utf-8")).hexdigest(), cleaned_filename, language)
         with open(sol_filename, mode='w', encoding='utf8', newline='') as fi:
             fi.write(remainder)
 
-def extract_and_write(path):
+def extract_and_write(path, language):
```

Comment:
Since `language` is just a string and not an enum and because someone might try `None` instead of `""` (which would skip all cases), I'd add an assert here:
```python
assert language in ["solidity", "yul", ""]
```

Reply:
Even though the command line parser is checking it already?

Reply:
That's an input validation. `assert`s are sanity checks for stuff that should never actually happen. Here it would guard against someone modifying the script and mistakenly passing in something unexpected. I'm really only suggesting it because consequences of such a mistake would be pretty bad (silently skipping all file types).